src/sql/workbench/parts/objectExplorer/common/objectExplorerViewTreeShim.ts

```diff
@@ -154,6 +157,31 @@ export class OEShimService extends Disposable implements IOEShimService {
 			}
 		}
 		icon = icon.toLowerCase();
+		// Change the database if the node has a different database
+		// than its parent
+		let databaseChanged = false;
+		let updatedPayload: azdata.IConnectionProfile | any = {};
+		if (node.nodeTypeId === NodeType.Database) {
+			const database = node.getDatabaseName();
+			if (database) {
+				databaseChanged = true;
```

Comment:
You should also check that the database isn't the same as the parentNode.payload.databaseName

Reply:
That's the bug we're fixing here. It's always the same, which needs to be updated from the metadata.